runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       polling_state = PollingState.new(azure_response, @long_running_operation_retry_timeout)
       request = azure_response.request
 
-      if !AsyncOperationStatus.is_terminal_status(polling_state.status)
+      while !AsyncOperationStatus.is_terminal_status(polling_state.status)
```

Comment:
Could you please elaborate on why we need while. If i understand correctly, then `Concurrent::TimerTask` should do polling unitl success on interval of `polling_delay`